pyspeckit/spectrum/units.py

```diff
@@ -354,80 +315,63 @@ def __new__(self, xarr, unit="Hz", frame='rest', frame_offset=0.0,
         subarr = np.array(xarr,dtype=dtype)
         subarr = subarr.view(self)
 
-        # allow either plural or singular unit
-        if units is not None and units in unit_type_dict and unit=='Hz':
-            unit=units
-
-        if unit is not None and " " in unit:
-            unit = unit.replace(" ","")
-
-        if unit in unit_type_dict:
-            subarr.units = unit
-        else:
-            if bad_unit_response=='raise':
-                raise ValueError('Unit %s not recognized.' % unit)
-            elif bad_unit_response=="pixel":
-                subarr.units="unknown"
-            else:
-                raise ValueError("Invalied bad unit response.  Valid options are 'raise','pixel'")
-        if subarr.units is None:
-            subarr.units = 'none'
-        subarr.frame = frame
-
-        # not currently used / placeholder
-        subarr.frame_offset = frame_offset
-
-        if xtype in xtype_dict:
-            subarr.xtype = xtype_dict[xtype]
-            subarr.frame = frame_dict[xtype]
-        elif subarr.units in unit_type_dict:
-            subarr.xtype = unit_type_dict[subarr.units]
-        elif type(xtype) is str:
-            warnings.warn("Unknown X-axis type in header: %s" % xtype)
-            subarr.xtype = xtype
-        else:
-            if xtype is not None:
-                warnings.warn("WARNING: xtype has been specified but was not recognized as a string.")
-            subarr.xtype = 'unknown'
-
+        subarr._unit = self.validate_unit(unit, bad_unit_response)
         subarr.refX = refX
+
         if refX_units is None:
-            if subarr.units in frequency_dict:
-                subarr.refX_units = subarr.units
+            if subarr._unit in frequency_dict:
+                refX_units = subarr.unit
             else:
-                subarr.refX_units = 'Hz'
+                refX_units = 'Hz'
+            subarr.refX_units = refX_units
```

Comment:
but we should really allow the plural version and start putting up deprecation warnings when it's used